lib/boavizta.ts

```diff
@@ -0,0 +1,135 @@
+import {IImpactModelInterface, IStaticParams} from "./interfaces/index";
+import axios from "axios";
+
+export {
+    IImpactModelInterface,
+    IStaticParams
+} from "./interfaces/index";
+
+
+export interface IBoaviztaStaticParams extends IStaticParams {
+    provider?: string;
+    componentType?: string;
+}
+
+export interface IBoaviztaCloudTelemetry {
+
+}
+
+
+export interface IBoaviztaAuthCredentials {
+}
+
+export class BoaviztaCloudImpactModel implements IImpactModelInterface {
+    public provider: string | undefined;
+    public name: string | undefined;
+    protected authCredentials: IBoaviztaAuthCredentials | undefined;
+
+    create(name: string, staticParams: object | undefined): object {
+        const staticParamCast = staticParams as IBoaviztaStaticParams;
```

Comment:
We've discussed and it's documentation in the issues, just to use `object` for now to keep things simple.